Acc tests output : 

New testcase added from this PR : 
```console
make t T=TestAccACMCertificate_PrivateKeyWo K=acm
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f/acm-certificate-private-key-wo 🌿...
TF_ACC=1 go1.26.3 test ./internal/service/acm/... -v -count 1 -parallel 20 -run='TestAccACMCertificate_PrivateKeyWo'  -timeout 360m -vet=off
2026/05/23 13:37:59 Creating Terraform AWS Provider (SDKv2-style)...
2026/05/23 13:37:59 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccACMCertificate_PrivateKeyWo
=== PAUSE TestAccACMCertificate_PrivateKeyWo
=== CONT  TestAccACMCertificate_PrivateKeyWo
--- PASS: TestAccACMCertificate_PrivateKeyWo (42.42s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/acm        50.673s
```

Existing tests : 

Few failed but seems to the same case in upstream branch.

```console
make t T=TestAccACMCertificate_ K=acm            
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 f/acm-certificate-private-key-wo 🌿...
TF_ACC=1 go1.26.3 test ./internal/service/acm/... -v -count 1 -parallel 20 -run='TestAccACMCertificate_'  -timeout 360m -vet=off
2026/05/23 21:12:21 Creating Terraform AWS Provider (SDKv2-style)...
2026/05/23 21:12:21 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccACMCertificate_Identity_basic
=== PAUSE TestAccACMCertificate_Identity_basic
=== RUN   TestAccACMCertificate_Identity_regionOverride
=== PAUSE TestAccACMCertificate_Identity_regionOverride
=== RUN   TestAccACMCertificate_Identity_ExistingResource_basic
=== PAUSE TestAccACMCertificate_Identity_ExistingResource_basic
=== RUN   TestAccACMCertificate_Identity_ExistingResource_noRefreshNoChange
=== PAUSE TestAccACMCertificate_Identity_ExistingResource_noRefreshNoChange
=== RUN   TestAccACMCertificate_tags
=== PAUSE TestAccACMCertificate_tags
=== RUN   TestAccACMCertificate_Tags_null
=== PAUSE TestAccACMCertificate_Tags_null
=== RUN   TestAccACMCertificate_Tags_emptyMap
=== PAUSE TestAccACMCertificate_Tags_emptyMap
=== RUN   TestAccACMCertificate_Tags_addOnUpdate
=== PAUSE TestAccACMCertificate_Tags_addOnUpdate
=== RUN   TestAccACMCertificate_Tags_EmptyTag_onCreate
=== PAUSE TestAccACMCertificate_Tags_EmptyTag_onCreate
=== RUN   TestAccACMCertificate_Tags_EmptyTag_OnUpdate_add
=== PAUSE TestAccACMCertificate_Tags_EmptyTag_OnUpdate_add
=== RUN   TestAccACMCertificate_Tags_EmptyTag_OnUpdate_replace
=== PAUSE TestAccACMCertificate_Tags_EmptyTag_OnUpdate_replace
=== RUN   TestAccACMCertificate_Tags_DefaultTags_providerOnly
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_providerOnly
=== RUN   TestAccACMCertificate_Tags_DefaultTags_nonOverlapping
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_nonOverlapping
=== RUN   TestAccACMCertificate_Tags_DefaultTags_overlapping
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_overlapping
=== RUN   TestAccACMCertificate_Tags_DefaultTags_updateToProviderOnly
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_updateToProviderOnly
=== RUN   TestAccACMCertificate_Tags_DefaultTags_updateToResourceOnly
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_updateToResourceOnly
=== RUN   TestAccACMCertificate_Tags_DefaultTags_emptyResourceTag
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_emptyResourceTag
=== RUN   TestAccACMCertificate_Tags_DefaultTags_emptyProviderOnlyTag
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_emptyProviderOnlyTag
=== RUN   TestAccACMCertificate_Tags_DefaultTags_nullOverlappingResourceTag
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_nullOverlappingResourceTag
=== RUN   TestAccACMCertificate_Tags_DefaultTags_nullNonOverlappingResourceTag
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_nullNonOverlappingResourceTag
=== RUN   TestAccACMCertificate_Tags_ComputedTag_onCreate
=== PAUSE TestAccACMCertificate_Tags_ComputedTag_onCreate
=== RUN   TestAccACMCertificate_Tags_ComputedTag_OnUpdate_add
=== PAUSE TestAccACMCertificate_Tags_ComputedTag_OnUpdate_add
=== RUN   TestAccACMCertificate_Tags_ComputedTag_OnUpdate_replace
=== PAUSE TestAccACMCertificate_Tags_ComputedTag_OnUpdate_replace
=== RUN   TestAccACMCertificate_Tags_IgnoreTags_Overlap_defaultTag
=== PAUSE TestAccACMCertificate_Tags_IgnoreTags_Overlap_defaultTag
=== RUN   TestAccACMCertificate_Tags_IgnoreTags_Overlap_resourceTag
=== PAUSE TestAccACMCertificate_Tags_IgnoreTags_Overlap_resourceTag
=== RUN   TestAccACMCertificate_emailValidation
=== PAUSE TestAccACMCertificate_emailValidation
=== RUN   TestAccACMCertificate_dnsValidation
=== PAUSE TestAccACMCertificate_dnsValidation
=== RUN   TestAccACMCertificate_root
=== PAUSE TestAccACMCertificate_root
=== RUN   TestAccACMCertificate_validationOptions
=== PAUSE TestAccACMCertificate_validationOptions
=== RUN   TestAccACMCertificate_privateCertificate_renewable
=== PAUSE TestAccACMCertificate_privateCertificate_renewable
=== RUN   TestAccACMCertificate_privateCertificate_noRenewalPermission
=== PAUSE TestAccACMCertificate_privateCertificate_noRenewalPermission
=== RUN   TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
=== PAUSE TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
=== RUN   TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
=== PAUSE TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
=== RUN   TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
=== PAUSE TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
=== RUN   TestAccACMCertificate_privateCertificate_addEarlyRenewalPastIneligible
=== PAUSE TestAccACMCertificate_privateCertificate_addEarlyRenewalPastIneligible
=== RUN   TestAccACMCertificate_privateCertificate_addEarlyRenewalFuture
=== PAUSE TestAccACMCertificate_privateCertificate_addEarlyRenewalFuture
=== RUN   TestAccACMCertificate_privateCertificate_updateEarlyRenewalFuture
=== PAUSE TestAccACMCertificate_privateCertificate_updateEarlyRenewalFuture
=== RUN   TestAccACMCertificate_privateCertificate_removeEarlyRenewal
=== PAUSE TestAccACMCertificate_privateCertificate_removeEarlyRenewal
=== RUN   TestAccACMCertificate_Root_trailingPeriod
=== PAUSE TestAccACMCertificate_Root_trailingPeriod
=== RUN   TestAccACMCertificate_rootAndWildcardSan
=== PAUSE TestAccACMCertificate_rootAndWildcardSan
=== RUN   TestAccACMCertificate_SubjectAlternativeNames_emptyString
=== PAUSE TestAccACMCertificate_SubjectAlternativeNames_emptyString
=== RUN   TestAccACMCertificate_San_single
=== PAUSE TestAccACMCertificate_San_single
=== RUN   TestAccACMCertificate_San_multiple
=== PAUSE TestAccACMCertificate_San_multiple
=== RUN   TestAccACMCertificate_San_trailingPeriod
=== PAUSE TestAccACMCertificate_San_trailingPeriod
=== RUN   TestAccACMCertificate_San_matches_domain
=== PAUSE TestAccACMCertificate_San_matches_domain
=== RUN   TestAccACMCertificate_wildcard
=== PAUSE TestAccACMCertificate_wildcard
=== RUN   TestAccACMCertificate_wildcardAndRootSan
=== PAUSE TestAccACMCertificate_wildcardAndRootSan
=== RUN   TestAccACMCertificate_keyAlgorithm
=== PAUSE TestAccACMCertificate_keyAlgorithm
=== RUN   TestAccACMCertificate_disableCTLogging
=== PAUSE TestAccACMCertificate_disableCTLogging
=== RUN   TestAccACMCertificate_disableReenableCTLogging
=== PAUSE TestAccACMCertificate_disableReenableCTLogging
=== RUN   TestAccACMCertificate_PrivateKeyWo
=== PAUSE TestAccACMCertificate_PrivateKeyWo
=== RUN   TestAccACMCertificate_Imported_domainName
=== PAUSE TestAccACMCertificate_Imported_domainName
=== RUN   TestAccACMCertificate_Imported_validityDates
=== PAUSE TestAccACMCertificate_Imported_validityDates
=== RUN   TestAccACMCertificate_Imported_ipAddress
=== PAUSE TestAccACMCertificate_Imported_ipAddress
=== RUN   TestAccACMCertificate_PrivateKey_ReimportWithTags
=== PAUSE TestAccACMCertificate_PrivateKey_ReimportWithTags
=== RUN   TestAccACMCertificate_optionExport
    certificate_test.go:1778: skipping test; environment variable ACM_TEST_CERTIFICATE_EXPORT must be set
--- SKIP: TestAccACMCertificate_optionExport (0.00s)
=== CONT  TestAccACMCertificate_Identity_basic
=== CONT  TestAccACMCertificate_validationOptions
=== CONT  TestAccACMCertificate_Tags_DefaultTags_updateToProviderOnly
=== CONT  TestAccACMCertificate_San_multiple
=== CONT  TestAccACMCertificate_Tags_addOnUpdate
=== CONT  TestAccACMCertificate_privateCertificate_addEarlyRenewalFuture
=== CONT  TestAccACMCertificate_Tags_DefaultTags_nullOverlappingResourceTag
=== CONT  TestAccACMCertificate_Tags_ComputedTag_onCreate
=== CONT  TestAccACMCertificate_Tags_DefaultTags_nullNonOverlappingResourceTag
=== CONT  TestAccACMCertificate_Imported_validityDates
=== CONT  TestAccACMCertificate_PrivateKey_ReimportWithTags
=== CONT  TestAccACMCertificate_Imported_ipAddress
=== CONT  TestAccACMCertificate_wildcardAndRootSan
=== CONT  TestAccACMCertificate_disableCTLogging
=== CONT  TestAccACMCertificate_keyAlgorithm
=== CONT  TestAccACMCertificate_emailValidation
=== CONT  TestAccACMCertificate_root
=== CONT  TestAccACMCertificate_disableReenableCTLogging
=== CONT  TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
=== CONT  TestAccACMCertificate_Tags_ComputedTag_OnUpdate_add
=== NAME  TestAccACMCertificate_disableReenableCTLogging
    certificate_test.go:1420: Step 1/7 error: Error running pre-apply plan: exit status 1
        
        Error: no matching Route 53 Hosted Zone found
        
          with data.aws_route53_zone.test,
          on terraform_plugin_test.tf line 21, in data "aws_route53_zone" "test":
          21: data "aws_route53_zone" "test" {
        
--- FAIL: TestAccACMCertificate_disableReenableCTLogging (6.66s)
=== CONT  TestAccACMCertificate_dnsValidation
=== NAME  TestAccACMCertificate_Imported_ipAddress
    certificate_test.go:1660: Step 1/2 error: Check failed: Check 2/4 error: aws_acm_certificate.test: Attribute 'domain_name' expected "", got "1.2.3.4"
        Check 4/4 error: aws_acm_certificate.test: Attribute 'subject_alternative_names.#' expected "0", got "1"
--- FAIL: TestAccACMCertificate_Imported_ipAddress (21.75s)
=== CONT  TestAccACMCertificate_privateCertificate_noRenewalPermission
--- PASS: TestAccACMCertificate_Imported_validityDates (35.89s)
=== CONT  TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
--- PASS: TestAccACMCertificate_Tags_DefaultTags_nullOverlappingResourceTag (44.19s)
=== CONT  TestAccACMCertificate_tags
--- PASS: TestAccACMCertificate_Tags_ComputedTag_onCreate (45.28s)
=== CONT  TestAccACMCertificate_Tags_emptyMap
--- PASS: TestAccACMCertificate_wildcardAndRootSan (50.86s)
=== CONT  TestAccACMCertificate_Tags_null
--- PASS: TestAccACMCertificate_root (50.92s)
=== CONT  TestAccACMCertificate_privateCertificate_renewable
--- PASS: TestAccACMCertificate_keyAlgorithm (51.57s)
=== CONT  TestAccACMCertificate_Identity_ExistingResource_basic
--- PASS: TestAccACMCertificate_San_multiple (51.61s)
=== CONT  TestAccACMCertificate_Identity_ExistingResource_noRefreshNoChange
--- PASS: TestAccACMCertificate_dnsValidation (45.87s)
=== CONT  TestAccACMCertificate_San_matches_domain
--- PASS: TestAccACMCertificate_disableCTLogging (52.58s)
=== CONT  TestAccACMCertificate_wildcard
--- PASS: TestAccACMCertificate_Tags_DefaultTags_nullNonOverlappingResourceTag (52.75s)
=== CONT  TestAccACMCertificate_privateCertificate_addEarlyRenewalPastIneligible
--- PASS: TestAccACMCertificate_validationOptions (59.72s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_providerOnly
--- PASS: TestAccACMCertificate_emailValidation (60.94s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_overlapping
--- PASS: TestAccACMCertificate_Identity_basic (61.84s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_nonOverlapping
=== NAME  TestAccACMCertificate_PrivateKey_ReimportWithTags
    certificate_test.go:1697: Step 4/4 error running import: ImportStateVerify attributes not equivalent. Difference is shown below. The - symbol indicates attributes missing after import.
        
          map[string]string{
        -       "subject_alternative_names.0": "1.2.3.4",
        +       "subject_alternative_names.0": "5.6.7.8",
          }
--- PASS: TestAccACMCertificate_Tags_addOnUpdate (66.37s)
=== CONT  TestAccACMCertificate_San_trailingPeriod
--- PASS: TestAccACMCertificate_Tags_DefaultTags_updateToProviderOnly (70.46s)
=== CONT  TestAccACMCertificate_Tags_IgnoreTags_Overlap_defaultTag
--- FAIL: TestAccACMCertificate_PrivateKey_ReimportWithTags (71.42s)
=== CONT  TestAccACMCertificate_Tags_IgnoreTags_Overlap_resourceTag
=== NAME  TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
    certificate_test.go:535: Step 3/4 error: Check failed: Check 2/11 error: ACM Certificate not renewed: i.NotAfter="2027-06-22 15:42:44 +0000 UTC", j.NotAfter="2027-06-22 15:42:44 +0000 UTC"
        Check 5/11 error: aws_acm_certificate.test: Attribute 'renewal_eligibility' expected "INELIGIBLE", got "ELIGIBLE"
        Check 6/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.#' expected "1", got "0"
        Check 7/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.renewal_status' not found
        Check 8/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.renewal_status_reason' not found
        Check 9/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.updated_at' didn't match "^[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\\.[0-9]+)?([Zz]|([+-]([01][0-9]|2[0-3]):[0-5][0-9]))$", got ""
--- PASS: TestAccACMCertificate_Tags_ComputedTag_OnUpdate_add (75.21s)
=== CONT  TestAccACMCertificate_rootAndWildcardSan
--- FAIL: TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration (81.87s)
=== CONT  TestAccACMCertificate_San_single
--- PASS: TestAccACMCertificate_wildcard (43.89s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_emptyResourceTag
--- PASS: TestAccACMCertificate_San_matches_domain (47.63s)
=== CONT  TestAccACMCertificate_SubjectAlternativeNames_emptyString
--- PASS: TestAccACMCertificate_SubjectAlternativeNames_emptyString (1.41s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_emptyProviderOnlyTag
--- PASS: TestAccACMCertificate_Tags_emptyMap (60.86s)
=== CONT  TestAccACMCertificate_Imported_domainName
=== NAME  TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
    certificate_test.go:460: Step 3/4 error: Check failed: Check 2/11 error: ACM Certificate not renewed: i.NotAfter="2027-06-22 15:43:15 +0000 UTC", j.NotAfter="2027-06-22 15:43:15 +0000 UTC"
        Check 5/11 error: aws_acm_certificate.test: Attribute 'renewal_eligibility' expected "INELIGIBLE", got "ELIGIBLE"
        Check 6/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.#' expected "1", got "0"
        Check 7/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.renewal_status' not found
        Check 8/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.renewal_status_reason' not found
        Check 9/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.updated_at' didn't match "^[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\\.[0-9]+)?([Zz]|([+-]([01][0-9]|2[0-3]):[0-5][0-9]))$", got ""
--- PASS: TestAccACMCertificate_Tags_null (59.99s)
=== CONT  TestAccACMCertificate_PrivateKeyWo
--- PASS: TestAccACMCertificate_San_trailingPeriod (46.29s)
=== CONT  TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
--- FAIL: TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration (83.29s)
=== CONT  TestAccACMCertificate_Tags_EmptyTag_OnUpdate_add
--- PASS: TestAccACMCertificate_rootAndWildcardSan (44.81s)
=== CONT  TestAccACMCertificate_privateCertificate_removeEarlyRenewal
--- PASS: TestAccACMCertificate_Identity_ExistingResource_noRefreshNoChange (70.94s)
=== CONT  TestAccACMCertificate_Tags_EmptyTag_OnUpdate_replace
--- PASS: TestAccACMCertificate_San_single (52.53s)
=== CONT  TestAccACMCertificate_Root_trailingPeriod
--- PASS: TestAccACMCertificate_Root_trailingPeriod (2.58s)
=== CONT  TestAccACMCertificate_privateCertificate_updateEarlyRenewalFuture
--- PASS: TestAccACMCertificate_privateCertificate_addEarlyRenewalFuture (137.92s)
=== CONT  TestAccACMCertificate_Tags_EmptyTag_onCreate
--- PASS: TestAccACMCertificate_Tags_DefaultTags_emptyResourceTag (46.66s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_updateToResourceOnly
--- PASS: TestAccACMCertificate_Tags_DefaultTags_emptyProviderOnlyTag (44.61s)
=== CONT  TestAccACMCertificate_Identity_regionOverride
--- PASS: TestAccACMCertificate_privateCertificate_addEarlyRenewalPastIneligible (100.86s)
=== CONT  TestAccACMCertificate_Tags_ComputedTag_OnUpdate_replace
--- PASS: TestAccACMCertificate_Identity_ExistingResource_basic (109.51s)
--- PASS: TestAccACMCertificate_Tags_IgnoreTags_Overlap_defaultTag (95.05s)
--- PASS: TestAccACMCertificate_PrivateKeyWo (58.05s)
--- PASS: TestAccACMCertificate_Tags_DefaultTags_nonOverlapping (114.79s)
--- PASS: TestAccACMCertificate_Tags_IgnoreTags_Overlap_resourceTag (105.37s)
--- PASS: TestAccACMCertificate_Tags_DefaultTags_overlapping (116.45s)
=== NAME  TestAccACMCertificate_Imported_domainName
    certificate_test.go:1566: Step 4/4 error running import: ImportStateVerify attributes not equivalent. Difference is shown below. The - symbol indicates attributes missing after import.
        
          map[string]string{
        -       "subject_alternative_names.0": "example.com",
        +       "subject_alternative_names.0": "m2iueei1.test",
          }
--- FAIL: TestAccACMCertificate_Imported_domainName (82.56s)
--- PASS: TestAccACMCertificate_Tags_EmptyTag_OnUpdate_replace (66.61s)
--- PASS: TestAccACMCertificate_tags (146.09s)
--- PASS: TestAccACMCertificate_Tags_DefaultTags_updateToResourceOnly (53.82s)
--- PASS: TestAccACMCertificate_Tags_DefaultTags_providerOnly (140.78s)
--- PASS: TestAccACMCertificate_Tags_EmptyTag_onCreate (64.35s)
--- PASS: TestAccACMCertificate_privateCertificate_renewable (151.70s)
--- PASS: TestAccACMCertificate_Identity_regionOverride (60.52s)
--- PASS: TestAccACMCertificate_privateCertificate_removeEarlyRenewal (89.83s)
--- PASS: TestAccACMCertificate_Tags_EmptyTag_OnUpdate_add (92.91s)
--- PASS: TestAccACMCertificate_privateCertificate_updateEarlyRenewalFuture (79.93s)
=== NAME  TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
    certificate_test.go:610: Step 5/5 error: Check failed: Check 4/11 error: aws_acm_certificate.test: Attribute 'pending_renewal' expected "false", got "true"
--- FAIL: TestAccACMCertificate_privateCertificate_addEarlyRenewalPast (122.70s)
--- PASS: TestAccACMCertificate_Tags_ComputedTag_OnUpdate_replace (85.25s)
--- PASS: TestAccACMCertificate_privateCertificate_noRenewalPermission (237.25s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/acm        265.869s
FAIL
make: *** [t] Error 1
```


In main branch : 

```console
make t T=TestAccACMCertificate_ K=acm
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
make: Running acceptance tests on branch: 🌿 main 🌿...
TF_ACC=1 go1.26.3 test ./internal/service/acm/... -v -count 1 -parallel 20 -run='TestAccACMCertificate_'  -timeout 360m -vet=off
2026/05/23 18:59:49 Creating Terraform AWS Provider (SDKv2-style)...
2026/05/23 18:59:49 Initializing Terraform AWS Provider (SDKv2-style)...
=== RUN   TestAccACMCertificate_Identity_basic
=== PAUSE TestAccACMCertificate_Identity_basic
=== RUN   TestAccACMCertificate_Identity_regionOverride
=== PAUSE TestAccACMCertificate_Identity_regionOverride
=== RUN   TestAccACMCertificate_Identity_ExistingResource_basic
=== PAUSE TestAccACMCertificate_Identity_ExistingResource_basic
=== RUN   TestAccACMCertificate_Identity_ExistingResource_noRefreshNoChange
=== PAUSE TestAccACMCertificate_Identity_ExistingResource_noRefreshNoChange
=== RUN   TestAccACMCertificate_tags
=== PAUSE TestAccACMCertificate_tags
=== RUN   TestAccACMCertificate_Tags_null
=== PAUSE TestAccACMCertificate_Tags_null
=== RUN   TestAccACMCertificate_Tags_emptyMap
=== PAUSE TestAccACMCertificate_Tags_emptyMap
=== RUN   TestAccACMCertificate_Tags_addOnUpdate
=== PAUSE TestAccACMCertificate_Tags_addOnUpdate
=== RUN   TestAccACMCertificate_Tags_EmptyTag_onCreate
=== PAUSE TestAccACMCertificate_Tags_EmptyTag_onCreate
=== RUN   TestAccACMCertificate_Tags_EmptyTag_OnUpdate_add
=== PAUSE TestAccACMCertificate_Tags_EmptyTag_OnUpdate_add
=== RUN   TestAccACMCertificate_Tags_EmptyTag_OnUpdate_replace
=== PAUSE TestAccACMCertificate_Tags_EmptyTag_OnUpdate_replace
=== RUN   TestAccACMCertificate_Tags_DefaultTags_providerOnly
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_providerOnly
=== RUN   TestAccACMCertificate_Tags_DefaultTags_nonOverlapping
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_nonOverlapping
=== RUN   TestAccACMCertificate_Tags_DefaultTags_overlapping
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_overlapping
=== RUN   TestAccACMCertificate_Tags_DefaultTags_updateToProviderOnly
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_updateToProviderOnly
=== RUN   TestAccACMCertificate_Tags_DefaultTags_updateToResourceOnly
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_updateToResourceOnly
=== RUN   TestAccACMCertificate_Tags_DefaultTags_emptyResourceTag
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_emptyResourceTag
=== RUN   TestAccACMCertificate_Tags_DefaultTags_emptyProviderOnlyTag
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_emptyProviderOnlyTag
=== RUN   TestAccACMCertificate_Tags_DefaultTags_nullOverlappingResourceTag
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_nullOverlappingResourceTag
=== RUN   TestAccACMCertificate_Tags_DefaultTags_nullNonOverlappingResourceTag
=== PAUSE TestAccACMCertificate_Tags_DefaultTags_nullNonOverlappingResourceTag
=== RUN   TestAccACMCertificate_Tags_ComputedTag_onCreate
=== PAUSE TestAccACMCertificate_Tags_ComputedTag_onCreate
=== RUN   TestAccACMCertificate_Tags_ComputedTag_OnUpdate_add
=== PAUSE TestAccACMCertificate_Tags_ComputedTag_OnUpdate_add
=== RUN   TestAccACMCertificate_Tags_ComputedTag_OnUpdate_replace
=== PAUSE TestAccACMCertificate_Tags_ComputedTag_OnUpdate_replace
=== RUN   TestAccACMCertificate_Tags_IgnoreTags_Overlap_defaultTag
=== PAUSE TestAccACMCertificate_Tags_IgnoreTags_Overlap_defaultTag
=== RUN   TestAccACMCertificate_Tags_IgnoreTags_Overlap_resourceTag
=== PAUSE TestAccACMCertificate_Tags_IgnoreTags_Overlap_resourceTag
=== RUN   TestAccACMCertificate_emailValidation
=== PAUSE TestAccACMCertificate_emailValidation
=== RUN   TestAccACMCertificate_dnsValidation
=== PAUSE TestAccACMCertificate_dnsValidation
=== RUN   TestAccACMCertificate_root
=== PAUSE TestAccACMCertificate_root
=== RUN   TestAccACMCertificate_validationOptions
=== PAUSE TestAccACMCertificate_validationOptions
=== RUN   TestAccACMCertificate_privateCertificate_renewable
=== PAUSE TestAccACMCertificate_privateCertificate_renewable
=== RUN   TestAccACMCertificate_privateCertificate_noRenewalPermission
=== PAUSE TestAccACMCertificate_privateCertificate_noRenewalPermission
=== RUN   TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
=== PAUSE TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
=== RUN   TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
=== PAUSE TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
=== RUN   TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
=== PAUSE TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
=== RUN   TestAccACMCertificate_privateCertificate_addEarlyRenewalPastIneligible
=== PAUSE TestAccACMCertificate_privateCertificate_addEarlyRenewalPastIneligible
=== RUN   TestAccACMCertificate_privateCertificate_addEarlyRenewalFuture
=== PAUSE TestAccACMCertificate_privateCertificate_addEarlyRenewalFuture
=== RUN   TestAccACMCertificate_privateCertificate_updateEarlyRenewalFuture
=== PAUSE TestAccACMCertificate_privateCertificate_updateEarlyRenewalFuture
=== RUN   TestAccACMCertificate_privateCertificate_removeEarlyRenewal
=== PAUSE TestAccACMCertificate_privateCertificate_removeEarlyRenewal
=== RUN   TestAccACMCertificate_Root_trailingPeriod
=== PAUSE TestAccACMCertificate_Root_trailingPeriod
=== RUN   TestAccACMCertificate_rootAndWildcardSan
=== PAUSE TestAccACMCertificate_rootAndWildcardSan
=== RUN   TestAccACMCertificate_SubjectAlternativeNames_emptyString
=== PAUSE TestAccACMCertificate_SubjectAlternativeNames_emptyString
=== RUN   TestAccACMCertificate_San_single
=== PAUSE TestAccACMCertificate_San_single
=== RUN   TestAccACMCertificate_San_multiple
=== PAUSE TestAccACMCertificate_San_multiple
=== RUN   TestAccACMCertificate_San_trailingPeriod
=== PAUSE TestAccACMCertificate_San_trailingPeriod
=== RUN   TestAccACMCertificate_San_matches_domain
=== PAUSE TestAccACMCertificate_San_matches_domain
=== RUN   TestAccACMCertificate_wildcard
=== PAUSE TestAccACMCertificate_wildcard
=== RUN   TestAccACMCertificate_wildcardAndRootSan
=== PAUSE TestAccACMCertificate_wildcardAndRootSan
=== RUN   TestAccACMCertificate_keyAlgorithm
=== PAUSE TestAccACMCertificate_keyAlgorithm
=== RUN   TestAccACMCertificate_disableCTLogging
=== PAUSE TestAccACMCertificate_disableCTLogging
=== RUN   TestAccACMCertificate_disableReenableCTLogging
=== PAUSE TestAccACMCertificate_disableReenableCTLogging
=== RUN   TestAccACMCertificate_Imported_domainName
=== PAUSE TestAccACMCertificate_Imported_domainName
=== RUN   TestAccACMCertificate_Imported_validityDates
=== PAUSE TestAccACMCertificate_Imported_validityDates
=== RUN   TestAccACMCertificate_Imported_ipAddress
=== PAUSE TestAccACMCertificate_Imported_ipAddress
=== RUN   TestAccACMCertificate_PrivateKey_ReimportWithTags
=== PAUSE TestAccACMCertificate_PrivateKey_ReimportWithTags
=== RUN   TestAccACMCertificate_optionExport
    certificate_test.go:1737: skipping test; environment variable ACM_TEST_CERTIFICATE_EXPORT must be set
--- SKIP: TestAccACMCertificate_optionExport (0.00s)
=== CONT  TestAccACMCertificate_Identity_basic
=== CONT  TestAccACMCertificate_root
=== CONT  TestAccACMCertificate_San_single
=== CONT  TestAccACMCertificate_privateCertificate_addEarlyRenewalPastIneligible
=== CONT  TestAccACMCertificate_privateCertificate_updateEarlyRenewalFuture
=== CONT  TestAccACMCertificate_Root_trailingPeriod
=== CONT  TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
=== CONT  TestAccACMCertificate_disableCTLogging
=== CONT  TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
=== CONT  TestAccACMCertificate_privateCertificate_renewable
=== CONT  TestAccACMCertificate_privateCertificate_noRenewalPermission
=== CONT  TestAccACMCertificate_validationOptions
=== CONT  TestAccACMCertificate_Tags_DefaultTags_updateToProviderOnly
=== CONT  TestAccACMCertificate_dnsValidation
=== CONT  TestAccACMCertificate_emailValidation
=== CONT  TestAccACMCertificate_Tags_IgnoreTags_Overlap_resourceTag
=== CONT  TestAccACMCertificate_Tags_IgnoreTags_Overlap_defaultTag
=== CONT  TestAccACMCertificate_Tags_ComputedTag_OnUpdate_replace
=== CONT  TestAccACMCertificate_Tags_ComputedTag_OnUpdate_add
=== CONT  TestAccACMCertificate_Tags_ComputedTag_onCreate
--- PASS: TestAccACMCertificate_Root_trailingPeriod (6.50s)
=== CONT  TestAccACMCertificate_wildcard
--- PASS: TestAccACMCertificate_validationOptions (66.45s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_nullNonOverlappingResourceTag
--- PASS: TestAccACMCertificate_root (70.02s)
=== CONT  TestAccACMCertificate_keyAlgorithm
--- PASS: TestAccACMCertificate_San_single (70.89s)
=== CONT  TestAccACMCertificate_wildcardAndRootSan
--- PASS: TestAccACMCertificate_wildcard (64.48s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_nullOverlappingResourceTag
--- PASS: TestAccACMCertificate_disableCTLogging (71.00s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_emptyProviderOnlyTag
--- PASS: TestAccACMCertificate_Tags_ComputedTag_onCreate (71.08s)
=== CONT  TestAccACMCertificate_Imported_validityDates
--- PASS: TestAccACMCertificate_emailValidation (74.28s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_emptyResourceTag
--- PASS: TestAccACMCertificate_dnsValidation (90.50s)
=== CONT  TestAccACMCertificate_PrivateKey_ReimportWithTags
--- PASS: TestAccACMCertificate_Identity_basic (97.06s)
=== CONT  TestAccACMCertificate_Imported_ipAddress
=== NAME  TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
    certificate_test.go:460: Step 3/4 error: Check failed: Check 2/11 error: ACM Certificate not renewed: i.NotAfter="2027-06-22 13:30:19 +0000 UTC", j.NotAfter="2027-06-22 13:30:19 +0000 UTC"
        Check 5/11 error: aws_acm_certificate.test: Attribute 'renewal_eligibility' expected "INELIGIBLE", got "ELIGIBLE"
        Check 6/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.#' expected "1", got "0"
        Check 7/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.renewal_status' not found
        Check 8/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.renewal_status_reason' not found
        Check 9/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.updated_at' didn't match "^[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\\.[0-9]+)?([Zz]|([+-]([01][0-9]|2[0-3]):[0-5][0-9]))$", got ""
--- PASS: TestAccACMCertificate_Tags_DefaultTags_updateToProviderOnly (116.44s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_updateToResourceOnly
--- PASS: TestAccACMCertificate_Tags_ComputedTag_OnUpdate_replace (119.59s)
=== CONT  TestAccACMCertificate_privateCertificate_addEarlyRenewalFuture
=== NAME  TestAccACMCertificate_Imported_ipAddress
    certificate_test.go:1619: Step 1/2 error: Check failed: Check 2/4 error: aws_acm_certificate.test: Attribute 'domain_name' expected "", got "1.2.3.4"
        Check 4/4 error: aws_acm_certificate.test: Attribute 'subject_alternative_names.#' expected "0", got "1"
--- FAIL: TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration (126.12s)
=== CONT  TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
--- PASS: TestAccACMCertificate_Tags_ComputedTag_OnUpdate_add (130.63s)
=== CONT  TestAccACMCertificate_Imported_domainName
--- FAIL: TestAccACMCertificate_Imported_ipAddress (38.56s)
=== CONT  TestAccACMCertificate_privateCertificate_removeEarlyRenewal
--- PASS: TestAccACMCertificate_Imported_validityDates (73.95s)
=== CONT  TestAccACMCertificate_disableReenableCTLogging
--- PASS: TestAccACMCertificate_keyAlgorithm (77.20s)
=== CONT  TestAccACMCertificate_Tags_addOnUpdate
--- PASS: TestAccACMCertificate_Tags_DefaultTags_nullNonOverlappingResourceTag (82.07s)
=== CONT  TestAccACMCertificate_SubjectAlternativeNames_emptyString
--- PASS: TestAccACMCertificate_Tags_DefaultTags_emptyProviderOnlyTag (79.66s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_overlapping
--- PASS: TestAccACMCertificate_wildcardAndRootSan (80.12s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_nonOverlapping
=== NAME  TestAccACMCertificate_disableReenableCTLogging
    certificate_test.go:1420: Step 1/7 error: Error running pre-apply plan: exit status 1
        
        Error: no matching Route 53 Hosted Zone found
        
          with data.aws_route53_zone.test,
          on terraform_plugin_test.tf line 21, in data "aws_route53_zone" "test":
          21: data "aws_route53_zone" "test" {
        
--- PASS: TestAccACMCertificate_SubjectAlternativeNames_emptyString (2.80s)
=== CONT  TestAccACMCertificate_rootAndWildcardSan
--- PASS: TestAccACMCertificate_Tags_DefaultTags_nullOverlappingResourceTag (80.41s)
=== CONT  TestAccACMCertificate_Tags_DefaultTags_providerOnly
--- FAIL: TestAccACMCertificate_disableReenableCTLogging (6.72s)
=== CONT  TestAccACMCertificate_San_trailingPeriod
--- PASS: TestAccACMCertificate_privateCertificate_addEarlyRenewalPastIneligible (153.42s)
=== CONT  TestAccACMCertificate_Tags_EmptyTag_OnUpdate_replace
--- PASS: TestAccACMCertificate_Tags_IgnoreTags_Overlap_defaultTag (154.82s)
=== CONT  TestAccACMCertificate_San_matches_domain
--- PASS: TestAccACMCertificate_Tags_DefaultTags_emptyResourceTag (83.52s)
=== CONT  TestAccACMCertificate_Tags_EmptyTag_OnUpdate_add
--- PASS: TestAccACMCertificate_privateCertificate_updateEarlyRenewalFuture (165.88s)
=== CONT  TestAccACMCertificate_Tags_EmptyTag_onCreate
--- PASS: TestAccACMCertificate_Tags_IgnoreTags_Overlap_resourceTag (179.78s)
=== CONT  TestAccACMCertificate_tags
=== NAME  TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
    certificate_test.go:610: Step 5/5 error: Check failed: Check 4/11 error: aws_acm_certificate.test: Attribute 'pending_renewal' expected "false", got "true"
=== NAME  TestAccACMCertificate_privateCertificate_renewable
    certificate_test.go:212: Step 6/7 error: Check failed: Check 2/10 error: ACM Certificate not renewed: i.NotAfter="2027-06-22 13:32:24 +0000 UTC", j.NotAfter="2027-06-22 13:32:24 +0000 UTC"
=== NAME  TestAccACMCertificate_PrivateKey_ReimportWithTags
    certificate_test.go:1656: Step 4/4 error running import: ImportStateVerify attributes not equivalent. Difference is shown below. The - symbol indicates attributes missing after import.
        
          map[string]string{
        -       "subject_alternative_names.0": "1.2.3.4",
        +       "subject_alternative_names.0": "5.6.7.8",
          }
--- FAIL: TestAccACMCertificate_privateCertificate_addEarlyRenewalPast (208.63s)
=== CONT  TestAccACMCertificate_Identity_ExistingResource_basic
--- FAIL: TestAccACMCertificate_privateCertificate_renewable (216.45s)
=== CONT  TestAccACMCertificate_Tags_emptyMap
--- FAIL: TestAccACMCertificate_PrivateKey_ReimportWithTags (126.64s)
=== CONT  TestAccACMCertificate_Identity_ExistingResource_noRefreshNoChange
--- PASS: TestAccACMCertificate_rootAndWildcardSan (65.88s)
=== CONT  TestAccACMCertificate_Tags_null
--- PASS: TestAccACMCertificate_San_matches_domain (65.04s)
=== CONT  TestAccACMCertificate_San_multiple
--- PASS: TestAccACMCertificate_San_trailingPeriod (68.32s)
=== CONT  TestAccACMCertificate_Identity_regionOverride
=== NAME  TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
    certificate_test.go:535: Step 3/4 error: Check failed: Check 2/11 error: ACM Certificate not renewed: i.NotAfter="2027-06-22 13:32:22 +0000 UTC", j.NotAfter="2027-06-22 13:32:22 +0000 UTC"
        Check 5/11 error: aws_acm_certificate.test: Attribute 'renewal_eligibility' expected "INELIGIBLE", got "ELIGIBLE"
        Check 6/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.#' expected "1", got "0"
        Check 7/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.renewal_status' not found
        Check 8/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.renewal_status_reason' not found
        Check 9/11 error: aws_acm_certificate.test: Attribute 'renewal_summary.0.updated_at' didn't match "^[0-9]{4}-(0[1-9]|1[012])-(0[1-9]|[12][0-9]|3[01])[Tt]([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\\.[0-9]+)?([Zz]|([+-]([01][0-9]|2[0-3]):[0-5][0-9]))$", got ""
--- PASS: TestAccACMCertificate_Tags_DefaultTags_updateToResourceOnly (109.96s)
--- FAIL: TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration (104.61s)
=== NAME  TestAccACMCertificate_Imported_domainName
    certificate_test.go:1525: Step 4/4 error running import: ImportStateVerify attributes not equivalent. Difference is shown below. The - symbol indicates attributes missing after import.
        
          map[string]string{
        -       "subject_alternative_names.0": "example.com",
        +       "subject_alternative_names.0": "x2vlntgp.test",
          }
--- PASS: TestAccACMCertificate_Tags_addOnUpdate (107.02s)
--- PASS: TestAccACMCertificate_Tags_EmptyTag_OnUpdate_replace (106.32s)
--- FAIL: TestAccACMCertificate_Imported_domainName (134.26s)
--- PASS: TestAccACMCertificate_privateCertificate_removeEarlyRenewal (142.85s)
--- PASS: TestAccACMCertificate_San_multiple (59.47s)
--- PASS: TestAccACMCertificate_Tags_EmptyTag_onCreate (114.74s)
--- PASS: TestAccACMCertificate_privateCertificate_addEarlyRenewalFuture (168.33s)
--- PASS: TestAccACMCertificate_Tags_null (78.31s)
--- PASS: TestAccACMCertificate_privateCertificate_noRenewalPermission (296.66s)
--- PASS: TestAccACMCertificate_Tags_emptyMap (80.37s)
--- PASS: TestAccACMCertificate_Identity_ExistingResource_noRefreshNoChange (82.23s)
--- PASS: TestAccACMCertificate_Tags_EmptyTag_OnUpdate_add (143.94s)
--- PASS: TestAccACMCertificate_Tags_DefaultTags_nonOverlapping (155.06s)
--- PASS: TestAccACMCertificate_Identity_regionOverride (86.91s)
--- PASS: TestAccACMCertificate_Tags_DefaultTags_overlapping (159.03s)
--- PASS: TestAccACMCertificate_Identity_ExistingResource_basic (113.40s)
--- PASS: TestAccACMCertificate_Tags_DefaultTags_providerOnly (176.37s)
--- PASS: TestAccACMCertificate_tags (158.77s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/acm        348.642s
FAIL
make: *** [t] Error 1
```


Feature Branch - `f/acm-certificate-private-key-wo`

Total Failures: 7

Failed tests:

```console
TestAccACMCertificate_disableReenableCTLogging
TestAccACMCertificate_Imported_ipAddress
TestAccACMCertificate_PrivateKey_ReimportWithTags
TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
TestAccACMCertificate_Imported_domainName
TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
```

Main Branch - `main`

Total Failures: 8

Failed tests:

```console
TestAccACMCertificate_privateCertificate_pendingRenewalGoDuration
TestAccACMCertificate_Imported_ipAddress
TestAccACMCertificate_disableReenableCTLogging
TestAccACMCertificate_privateCertificate_addEarlyRenewalPast
TestAccACMCertificate_privateCertificate_renewable
TestAccACMCertificate_PrivateKey_ReimportWithTags
TestAccACMCertificate_privateCertificate_pendingRenewalRFC3339Duration
TestAccACMCertificate_Imported_domainName
```

So not related to this change.

--- a/internal/service/acm/certificate.go
+++ b/internal/service/acm/certificate.go
@@ -404,6 +404,9 @@ func resourceCertificateCreate(ctx context.Context, d *schema.ResourceData, meta
 		privateKey := d.Get(names.AttrPrivateKey).(string)
 		privateKeyWo, di := flex.GetWriteOnlyStringValue(d, cty.GetAttrPath("private_key_wo"))
 		diags = append(diags, di...)
+		if diags.HasError() {
+			return diags
+		}
 		if privateKeyWo != "" {
 			privateKey = privateKeyWo
 		}
@@ -518,6 +521,9 @@ func resourceCertificateUpdate(ctx context.Context, d *schema.ResourceData, meta
 			privateKey := d.Get(names.AttrPrivateKey).(string)
 			privateKeyWo, di := flex.GetWriteOnlyStringValue(d, cty.GetAttrPath("private_key_wo"))
 			diags = append(diags, di...)
+			if diags.HasError() {
+				return diags
+			}
 			if privateKeyWo != "" {
 				privateKey = privateKeyWo
 			}

